src/main/java/net/porillo/listeners/CO2Listener.java

```diff
@@ -87,37 +90,40 @@ public void onFurnaceSmelt(FurnaceBurnEvent event) {
 			FurnaceInsertQuery insertQuery = new FurnaceInsertQuery(furnace);
 			AsyncDBQueue.getInstance().queueInsertQuery(insertQuery);
 
-			gw.getLogger().warning(event.getFuel().getType().name() + " burned as fuel in an untracked furnace!");
-			gw.getLogger().warning("@ " + location.toString());
+			gw.getLogger().warning(String.format("[%s] burned as fuel in an untracked furnace!", event.getFuel().getType().name()));
+			gw.getLogger().warning(String.format("@ %s", location.toString()));
 		}
 
-		// Update the carbon score in the associated world
-		String associatedWorldName = eventClimateEngine.getConfig().getAssociation();
-		GWorld associatedWorld = GlobalWarming.getInstance().getTableManager().getWorldTable().getWorld(associatedWorldName);
+		//Carbon scores:
 		Contribution contrib = eventClimateEngine.furnaceBurn(furnace, event.getFuel());
+		if (contrib != null) {
```

Comment:
When would the contribution object be null? Is it when players logoff before the event is triggered?

Reply:
It's just a precaution (as things are, it shouldn't return null)